objects.js

```diff
@@ -2640,87 +2640,137 @@ SpriteMorph.prototype.reporterize = function (expressionString) {
     // highly experimental Christmas Easter Egg 2016 :-)
     var ast;
 
-    function parseInfix(expression, operator, already) {
-        // very basic binary infix parser for arithmetic expressions
-        // with strict left-to-right operator precedence (like in Smalltalk)
-        // which can be overriden by - nested - parentheses.
-        // assumes well-formed expressions, no graceful error handling yet.
-
-        var inputs = ['', ''],
-            idx = 0,
-            ch;
-
-        function format(value) {
-            return value instanceof Array || isNaN(+value) ? value : +value;
-        }
-
-        function nested() {
-            var level = 1,
-                expr = '';
-            while (idx < expression.length) {
-                ch = expression[idx];
-                idx += 1;
-                switch (ch) {
-                case '(':
-                    level += 1;
-                    break;
-                case ')':
-                    level -= 1;
-                    if (!level) {
-                        return expr;
-                    }
-                    break;
-                }
-                expr += ch;
+    function tokenize(expressionString) {
+        var tokens = [];
+        var token;
+        expressionString = expressionString.trim();
+        function findIdent() {
+            var next = "";
+            while (expressionString.length > 0 && /[A-Za-z_0-9]/.test(expressionString[0])) {
+                next += expressionString[0];
+                expressionString = expressionString.slice(1);
+            }
+            return next;
+        }
+        function findNumber() {
+            var next = "";
+            while (expressionString.length > 0 && /\d|\./.test(expressionString[0])) {
```

Comment:
(Apparently, I never hit post on this 'review'...you've already addressed some of the comments so sorry for the duplication. 

Sorry for not being clear, what I meant was an explicit index variable. (Maybe `position` or something, that is just an int.)
Then each loop would simply do `position += 1` and checking `expressionString[0]` would be `expressionString[position]`.

I don't know if it really makes a difference though, but I imagine with a long string and parsing on every change it could be noticeable especially on slower computers.